File: app/src/main/java/hu/epam/worktime/mvvmworkdroid/modules/details/viewmodel/DetailsViewModel.java
package hu.epam.worktime.mvvmworkdroid.modules.details.viewmodel;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import java.util.ArrayList;
import java.util.List;

import hu.epam.worktime.mvvmworkdroid.modules.details.model.DetailsModel;
import hu.epam.worktime.mvvmworkdroid.modules.details.router.DetailsRouter;
import hu.epam.worktime.mvvmworkdroid.modules.services.models.WorkTime;

/**
 * Created by Mihaly_Hunyady on 2017. 01. 06..
 */

public class DetailsViewModel extends BaseObservable {
    DetailsModel model;
    @Bindable
    private String arriveTime;
    @Bindable
    private String leaveTime;
    @Bindable
    private String currentDate;
    @Bindable
    private String dinnerStart;
    @Bindable
    private String dinnerEnd;
    private List<String> others = new ArrayList<>();

    public DetailsViewModel(DetailsModel model, DetailsRouter router) {
        this.model = model;
    }

    public void setSelectedWorkTime(WorkTime selectedWorkTime) {
        model.setSelectedWorkTime(selectedWorkTime);
        init();
    }

    private void init() {
        arriveTime = model.getArriveTime();
        leaveTime = model.getLeaveTime();
        currentDate = model.getCurrentDate();
        dinnerStart = model.getDinnerStartTime();
        dinnerEnd = model.getDinnerEndTime();
        notifyChange();
    }

    public String getArriveTime() {
        return arriveTime;
    }

       public String getLeaveTime() {
        return leaveTime;
    }

    public String getCurrentDate() {
        return currentDate;
    }

    public String getDinnerStart() {
        return dinnerStart;
    }

    public String getDinnerEnd() {
        return dinnerEnd;
    }

}

File: app/src/main/java/hu/epam/worktime/mvvmworkdroid/modules/details/viewmodel/DetailsViewModel.kt
package hu.epam.worktime.mvvmworkdroid.modules.details.viewmodel

import android.databinding.BaseObservable
import android.databinding.Bindable

import java.util.ArrayList

import hu.epam.worktime.mvvmworkdroid.modules.details.model.DetailsModel
import hu.epam.worktime.mvvmworkdroid.modules.details.router.DetailsRouter
import hu.epam.worktime.mvvmworkdroid.modules.services.models.WorkTime

/**
 * Created by Mihaly_Hunyady on 2017. 01. 06..
 */

class DetailsViewModel(internal var model: DetailsModel, router: DetailsRouter) : BaseObservable() {
    @Bindable
    var arriveTime: String? = null
        private set
    @Bindable
    var leaveTime: String? = null
        private set
    @Bindable
    var currentDate: String? = null
        private set
    @Bindable
    var dinnerStart: String? = null
        private set
    @Bindable
    var dinnerEnd: String? = null
        private set
    @Bindable
    private val others = ArrayList<String>()

    fun setSelectedWorkTime(selectedWorkTime: WorkTime) {
        model.setSelectedWorkTime(selectedWorkTime)
        init()
    }

    private fun init() {
        arriveTime = model.arriveTime
        leaveTime = model.leaveTime
        currentDate = model.currentDate
        dinnerStart = model.dinnerStartTime
        dinnerEnd = model.dinnerEndTime
        notifyChange()
    }

}
